pocs/camera/__init__.py

```diff
@@ -115,71 +115,73 @@ def kwargs_or_config(item, default=None):
+                    if connection_method not in device_config:
+                        raise error.CameraNotFound
+                except (KeyError, error.CameraNotFound):
                     raise error.CameraNotFound(
-                        msg="No port specified and auto_detect=False")
+                        msg=f"No connection_method for {model} specified and auto_detect=False")
 
-            camera_focuser = device_config.get('focuser', None)
-            camera_filterwheel = device_config.get('filterwheel', None)
-            camera_readout = device_config.get('readout_time', 6.0)
+            device_config.setdefault('focuser', None)
+            device_config.setdefault('filterwheel', None)
+            device_config.setdefault('readout_time', 6.0)
 
         else:
             logger.debug('Using camera simulator.')
             # Set up a simulated camera with fully configured simulated
             # focuser
-            camera_model = 'simulator'
-            camera_port = '/dev/camera/simulator'
-            camera_focuser = {'model': 'simulator',
-                              'focus_port': '/dev/ttyFAKE',
-                              'initial_position': 20000,
-                              'autofocus_range': (40, 80),
-                              'autofocus_step': (10, 20),
-                              'autofocus_seconds': 0.1,
-                              'autofocus_size': 500}
-            camera_filterwheel = {'model': 'simulator',
-                                  'filter_names': ['one', 'deux', 'drei', 'quattro'],
-                                  'move_time': 0.1 * u.second,
-                                  'timeout': 0.5 * u.second}
-            camera_readout = 0.5
-
-        camera_set_point = device_config.get('set_point', None)
-        camera_filter = device_config.get('filter_type', None)
-
-        logger.debug('Creating camera: {}'.format(camera_model))
+            device_config['model'] = 'simulator'
+            device_config['port'] = '/dev/camera/simulator'
+            device_config['focuser'] = {'model': 'simulator',
+                                        'focus_port': '/dev/ttyFAKE',
+                                        'initial_position': 20000,
+                                        'autofocus_range': (40, 80),
+                                        'autofocus_step': (10, 20),
+                                        'autofocus_seconds': 0.1,
+                                        'autofocus_size': 500}
+            device_config['filterwheel'] = {'model': 'simulator',
+                                            'filter_names': ['one', 'deux', 'drei', 'quattro'],
+                                            'move_time': 0.1 * u.second,
+                                            'timeout': 0.5 * u.second}
+            device_config['readout_time'] = 0.5
+
+        device_config.setdefault('set_point', None)
+        device_config.setdefault('filter_type', None)
```

Comment:
As above, not actually necessary to add default values for `set_point` and `filter_type` as the `Camera` classes already do sensible things when these arguments are absent.